src/lib/services/shared.ts

```diff
@@ -153,15 +165,64 @@ export function serviceRequest(
     headers,
   };
 
+  /**
+   * The request handler for the fetch call. This can be overridden by providing a
+   * `__callable` property in the `fetch` options.
+   */
+  let handler = _fetch;
   /* eslint-disable no-underscore-dangle */
   if (injectedFetchOptions?.__callable) {
+    handler = injectedFetchOptions.__callable.bind(this);
     /**
      * Remove the `__callable` property from the `fetch` options before passing the options along.
      */
     delete init.__callable;
-    return injectedFetchOptions.__callable.call(this, url, init);
   }
   /* eslint-enable no-underscore-dangle */
 
-  return _fetch(url, init);
+  /**
+   * If the resource is configured to prevent retries, there is no `manager` instance,
+   * or token, the request will be made as-is.
+   */
+  if (config.preventRetry || !manager || !token || !isRefreshToken(token)) {
+    return handler(url, init);
+  }
+
+  /**
+   * Automatic Retry Handling
+   */
+
+  const initialResponse = await handler(url, init);
+  /**
+   * If the response is "ok", we can return it as-is.
+   */
+  if (initialResponse.ok) {
+    return initialResponse;
+  }
+  /**
+   * Detect if the response might be caused by a token that
+   * has expired, and if so, attempt to refresh the token then
+   * retry the request.
+   */
+  const shouldAttemptTokenRefresh = initialResponse.status === 401;
```

Comment:
I assume this will get fleshed out with some GARE detection logic.

Reply:
Just added: https://github.com/globus/globus-sdk-javascript/pull/251/files#diff-2b95db59620b45a4ba4cad7c5431a1f4abe45cbdb50f8e82d9c0192f7668925eR203-R218

Which I think will be enough (for now...)